chat-microservice/app/main.py

```diff
@@ -27,3 +29,16 @@ def create_application() -> FastAPI:
 
 
 app = create_application()
+
+@app.middleware("http")
+async def request_middleware(request: Request, call_next):
+    Logging.info(f"Request started at {request.url.path}")
+    response = await call_next(request)
+    response_body = [chunk async for chunk in response.body_iterator]
+    response.body_iterator = iterate_in_threadpool(iter(response_body))
```

Comment:
I don't understand why do you assign `response.body_iterator` a value that you just got out of it